fxgl-samples/src/main/kotlin/sandbox/subscene/SubSceneSampleApp.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * FXGL - JavaFX Game Library. The MIT License (MIT).
+ * Copyright (c) AlmasB (almaslvl@gmail.com).
+ * See LICENSE for details.
+ */
+
+package sandbox.subscene
+
+import com.almasb.fxgl.app.GameApplication
+import com.almasb.fxgl.app.GameSettings
+import com.almasb.fxgl.dsl.FXGL
+import com.almasb.fxgl.scene.SubScene
+import javafx.event.EventHandler
+import javafx.util.Duration
+
+/**
+ * @author Serge Merzliakov (smerzlia@gmail.com)
+ */
+class SubSceneSampleApp : GameApplication() {
+
+
+	override fun initSettings(settings: GameSettings?) {
+		settings?.width = 700
+		settings?.height = 400
+		settings?.title = "SubScene Navigation Demo"
+		settings?.isMainMenuEnabled = false
+		settings?.isGameMenuEnabled = false
+		settings?.isIntroEnabled = false
+	}
+
+	override fun initGame() {
+
+		val handler: EventHandler<NavigateEvent> = EventHandler { e: NavigateEvent? ->
+			handleNavigate(e!!)
+		}
+
+		FXGL.getEventBus().addEventHandler(NAVIGATION, handler)
+	}
+
+	override fun initUI() {
+		// TODO - is this the correct way to set up a SubScene on startup
```

Comment:
At the moment, yes. I can't remember why -- I don't think it's a limitation, probably suboptimal handling of scenes during app initialization. Would be nice if we could push subscenes without having to wait for a new frame / tick.